pkg/autoscaler/aggregation/bucketing.go

```diff
@@ -52,6 +54,31 @@ func NewTimedFloat64Buckets(window, granularity time.Duration) *TimedFloat64Buck
 	}
 }
 
+// WindowTotal returns the sum of all valid buckets.
+func (t *TimedFloat64Buckets) WindowTotal(now time.Time) float64 {
+	now = now.Truncate(t.granularity)
+	t.bucketsMutex.RLock()
+	defer t.bucketsMutex.RUnlock()
+	switch d := now.Sub(t.lastWrite); {
+	case d <= 0:
+		// If LastWrite equal or greater than Now
+		// return the current WindowTotal.
+		return t.windowTotal
+	case d < t.window:
+		// If we haven't received metrics for some time, which is less than
```

Comment:
Should we escalate the lock in this case and actually adjust windowTotal/ditch the buckets? We could then share the code here with below's implementation in "record" and subsequent calls wouldn't need to do this all on their own?

Reply:
That was my initial idea, but for now I decided to go with read lock only -- minimal required access.